swingset/src/main/java/com/nqadmin/swingset/SSDataNavigator.java

```diff
@@ -886,23 +876,48 @@ protected void addToolTips() {
 
 	} // end protected void addToolTips() {
 	
+	/**
+	 * Common code to commit changes to the database from the rowset if
+	 * modifications are allowed. After commit is performs any
+	 * post-update operations.
+	 * 
+	 * @return true unless dBNav.allowUpdate() returns false
+	 * @throws SQLException SQL Exception if rowset call to updateRow() fails
+	 */
+	private boolean commitChangesIfAllowed() throws SQLException {
+		
+		boolean result = commitChangesIfAllowedWithoutPostUpdateOps();
+		
+		// After successful commit, perform any post-update operations
+		if (result) {
+			dBNav.performPostUpdateOps();
+		}
```

Comment:
Just to note: `dBNav.performPostUpdateOps()` is executed even if there are no rows in result set. I'd guess that doesn't matter.

Reply:
Good point. I think this is how it worked previously, but if you look at the JavaDoc for the default performPostUpdateOps(), it says, "Method to perform operations after the updateRow has been called." Well, updateRow() has not been called if there aren't any rows so I'm inclined to change this.

Reply:
@errael  Please have a look at my changes to commitChangesIfAllowedWithoutPostUpdateOps(). Now returns false if there are no records.

It still returns true if modification==false, but I guess that's OK. Feels like some cleanup is needed across SSDBNav.allowUpdate() and SSDataNavigator.setModification()/getModification().

Reply:
The `modification` variable and it's `get/set` methods stuff are inherited from pre 4.0. IIRC from my brief investigation working on 4.0, there was some situation where it was impermissible to modify the database; it may have even been for a specific database vendor. It may be that now this can be thrown out.

About returning false if there are no records. That means a bunch of code that used to be executed would no longer be executed, not just `performPostUpdateOps()`. Similarly, returning true if modification is false is pretty different. 

At this point in 4.0, I'd be inclined to make a minimal change to fix the bug. Then clean this area up in the next release.

Reply:
@errael @prasanthreddy-git 

Probably went too far, but decided to merge the two methods and add an argument to control post update operations.
Previously, if modification==false, commitChanges() returned true so I don't think we're changing that behavior.